src/components/input-field/examples/input-field-error-icon.tsx

```diff
@@ -7,25 +7,49 @@ const MIN_LENGTH = 6;
 })
 export class InputFieldErrorIconExample {
     @State()
-    private value;
+    private valueNative;
+    @State()
+    private valueConsumer;
 
     constructor() {
-        this.onChange = this.onChange.bind(this);
+        this.onChangeNative = this.onChangeNative.bind(this);
+        this.onChangeConsumer = this.onChangeConsumer.bind(this);
+        this.isInvalid = this.isInvalid.bind(this);
     }
 
     public render() {
-        return (
+        return [
             <limel-input-field
-                label="Text Field"
+                label="Text Field with native validation"
                 minlength={MIN_LENGTH}
                 helperText="Please enter at least 6 characters!"
-                value={this.value}
-                onChange={this.onChange}
-            />
-        );
+                value={this.valueNative}
+                onChange={this.onChangeNative}
+            />,
+            <limel-input-field
+                label="Text Field with consumer validation"
+                type={'email'}
+                invalid={this.isInvalid()}
+                helperText="Please enter an email with the domain 'test.com'"
+                value={this.valueConsumer}
+                onChange={this.onChangeConsumer}
+            />,
+        ];
     }
 
-    private onChange(event) {
-        this.value = event.detail;
+    private onChangeNative(event) {
+        this.valueNative = event.detail;
+    }
+    private onChangeConsumer(event) {
+        this.valueConsumer = event.detail;
+    }
+
+    private isInvalid() {
+        const substringLength = 9;
+        return !(
+            this.valueConsumer &&
+            this.valueConsumer.substr(-substringLength, substringLength) ===
+                '@test.com'
+        );
```

Comment:
I added this small improvement of the example validator.

If you think it makes the example harder to understand, feel free to veto it :)

Reply:
looks fine, I wasn't going for absolute perfection on the validator, just wanted to make a point ;) but if that is exacter then go with it